Makefile.pre.in

```diff
@@ -911,6 +911,21 @@ $(PYTHONFRAMEWORKDIR)/Versions/$(VERSION)/$(PYTHONFRAMEWORK): \
 	$(LN) -fsn Versions/Current/$(PYTHONFRAMEWORK) $(PYTHONFRAMEWORKDIR)/$(PYTHONFRAMEWORK)
 	$(LN) -fsn Versions/Current/Resources $(PYTHONFRAMEWORKDIR)/Resources
 
+# This rule is for iOS, which requires an annoyingly just slighly different
+# format for frameworks to macOS. It *doesn't* use a versioned framework, and
+# the Info.plist must be in the root of the framework.
+$(PYTHONFRAMEWORKDIR)/$(PYTHONFRAMEWORK): \
+		$(LIBRARY) \
+		$(RESSRCDIR)/Info.plist
+	$(INSTALL) -d -m $(DIRMODE) $(PYTHONFRAMEWORKDIR)
```

Comment:
Here as in other install-related rules modified or added, you don't support the use of the `DESTDIR` variable, which is fairly standard in many Makefiles and which is supported elsewhere in this Makefile including for macOS frameworks. Is there an explicit reason for not supporting `DESTDIR`? If not, I think we should since it is very commonly used by build scripts and would give _surprising_ results if silently not supported.

Reply:
As far as I understand it, the use case that makes `DESTDIR` interesting/useful doesn't exist with iOS.

My understanding is that `DESTDIR` is useful when you've got a configured project that has a prefix of `/usr/local` (or somewhere "system sensitive"), and you want to test/dry run a Makefile without *actually* installing into that system sensitive location. You can then use `make DESTDIR=/tmp/staging install`, and see what would actually be installed.

However, iOS builds are slightly different, because they *must* be relocatable, and they're *not* installed into a "system sensitive" location. To that end: 
* `PYTHONFRAMEWORKINSTALLNAMEPREFIX` is always an `@rpath`, so using `DESTDIR` in the `install_name` doesn't make any sense; and
* `configure` *must* be installed with `--enable-framework=<path>`, because the build must be a framework build, and there's no default system-wide install location (like `/Library/Frameworks/...`) that would really make sense. At best, we could make `--enable-framework=yes` default to somewhere in the `build` folder or (something similar); but even then, the install location isn't "system sensitive", so using `DESTDIR` as a mechanism to dry-run an install doesn't make sense (at least, not to me).

I'm happy to revisit this if you think it would be worth supporting `DESTDIR` for the second case for the sake of completeness; but my gut feel is that it's extra complexity that won't actually be useful in practice.

Reply:
I get it, and you get it, that, because we are building in the reasonable assumption that iOS frameworks will always be relocatable, you could always use the `--enable-framework=path` to point where you want the install to happen. But I think that would not be at all obvious to developers trying to use the `Makefile` that, for this one new case, `DESTDIR` does not work like it does for all other types of (non-Windows) Python builds and installs and how it works in most other projects' `Makefiles`. And, as it stands, if someone tried to use `DESTDIR`, they would silently get very unexpected results. If it did add a lot of complexity, that would be a factor but I don't think it really would: just pre-pending `$(DESTDIR)` to the small list of install targets similar to the existing macOS framework install rules. And I imagine that, if we don't, someone will come up with another reason for needing it ;)

Reply:
Fair enough. `DESTDIR` update incoming shortly.